rdmd_test.d

```diff
@@ -140,6 +141,18 @@ void runCompilerAgnosticTests(string rdmdApp, string defaultCompiler, string mod
     assert(res.status == 0, res.output);
     assert(res.output.canFind("Usage: rdmd [RDMD AND DMD OPTIONS]... program [PROGRAM OPTIONS]..."));
 
+    string helpText = res.output;
+
+    // verify help text matches expected defaultCompiler
+    {
+        enum compilerHelpLine = "  --compiler=comp    use the specified compiler (e.g. gdmd) instead of ";
+        auto offset = helpText.indexOf(compilerHelpLine);
+        assert(offset >= 0);
+        auto compilerInHelp = helpText[offset + compilerHelpLine.length .. $];
+        compilerInHelp = compilerInHelp[0 .. compilerInHelp.countUntil!isWhite];
```

Comment:
Don't mix slicing and `countUntil`! One counts code units, the other code points. It is always a mistake.

Either use `byCodeUnit` or just use `indexOf` (if we know the rest of the string, we should know what kind of whitespace it uses, too).

Reply:
So is `countUntil` performing autodecoding?

Reply:
Yes, `countUntil` operates on ranges, and strings are auto-decoded when interpreted as ranges.

Reply:
Also, do "end of line" sequences for WYSIWYG strings change or do they always match the source?

Reply:
Looking at dmd source, looks like they all normalize to `\n`

Reply:
> Yes, countUntil operates on ranges, and strings are auto-decoded when interpreted as ranges.

How odd...this means that indexes into a "string range" will differ from indexes into a "string".  I heard about this "autodecoding" confusion before but haven't run into it myself yet...illuminating